lib/div-alg.gi

```diff
@@ -2166,10 +2166,12 @@ local m,A,B,n1;
 
 #A:=SimpleComponentOfGroupRingByCharacter(F,G,n);
 B:=Irr(G);
-if IsCharacter(n) then
-  n1:=Position(Irr(G),n);
-else
+if IsPosInt(n) then
```

Comment:
Don't need `B` for `Irr(G)`, it's never used below.

Reply:
This is to avoid situations where DefiningCharacter... returns an error because the group has no character table yet.  It is used implicitly.